Content.Server/Chat/Managers/ChatSanitizationManager.cs

```diff
@@ -94,7 +95,7 @@ public bool TrySanitizeOutSmilies(string input, EntityUid speaker, out string sa
 
         foreach (var (smiley, replacement) in SmileyToEmote)
         {
-            if (input.EndsWith(smiley, true, CultureInfo.InvariantCulture))
+            if (input.EndsWith(smiley, true, CultureInfo.InvariantCulture) && (input.EndsWith(" "+smiley, true, CultureInfo.InvariantCulture) || input.Equals(smiley, StringComparison.InvariantCultureIgnoreCase)))
```

Comment:
Doing `" "+smiley` sucks because now you're allocating 60-something string instances when this function previous would do nothing in most cases. You're really just checking whether the character before the suffix is a space so you can just check that with something like `input[^(smiley.Length+1)] == " "`.

Reply:
Good point, the resulting conditions will probably be really ugly but still workable.